src/components/container/GlucoseChart/GlucoseChart.tsx

```diff
@@ -39,14 +43,16 @@ export default function (props: GlucoseChartProps) {
         }
 
         if (props.previewState) {
-            setGlucose(previewData(props.previewState, selectedDate).glucose)
-            setSteps(previewData(props.previewState, selectedDate).steps);
-            setSleepMinutes(previewData(props.previewState, selectedDate).sleepMinutes);
-            setLoading(false);
+            previewData(props.previewState, selectedDate).then(({ glucose, steps, sleepMinutes }) => {
+                setGlucose(glucose)
+                setSteps(steps);
+                setSleepMinutes(sleepMinutes);
+                setLoading(false);
+            });
             return;
         }
 
-        let glucoseReadingLoader = getGlucoseReadings(selectedDate);
+        let glucoseReadingLoader = glucoseContext?.readings ? getGlucoseReadingsFromContext() : getGlucoseReadings(selectedDate);
```

Comment:
I think we discussed this a bit on Slack - this is an attempt to prevent multiple coordinated widgets from all requesting the same data, right?  

I'm a little confused on exactly how this ends up working because it looks like `readings` isn't defaulted to an empty array in the `GlucoseDayCoordinator`?  So it seems like on initial load, `glucoseContext?.readings` would return false even if you're inside of a valid `GlucoseContext`?  So it looks like you'd still get a double load - what am I missing?

Reply:
It works because the `GlucoseDayCoordinator` does not render its children until it has finished loading its readings.